### ISSUE TITLE
NoMethodError: undefined method `provides' for Chef::Provider::DockerService::Execute:Class

### ISSUE THREAD

**Issue by josephsiefers** [bflad/chef-docker#330]
I'm getting the following error with 0.40.0. Am I missing something in my configuration?

NoMethodError
-------------
undefined method `provides' for Chef::Provider::DockerService::Execute:Class

Cookbook Trace:
---------------
/var/lib/aws/opsworks/cache.stage2/cookbooks/docker/libraries/provider_docker_service_execute.rb:5:in `<class:Execute>'
/var/lib/aws/opsworks/cache.stage2/cookbooks/docker/libraries/provider_docker_service_execute.rb:4:in `<class:DockerService>'
/var/lib/aws/opsworks/cache.stage2/cookbooks/docker/libraries/provider_docker_service_execute.rb:3:in `<class:Provider>'
/var/lib/aws/opsworks/cache.stage2/cookbooks/docker/libraries/provider_docker_service_execute.rb:2:in `<class:Chef>'
/var/lib/aws/opsworks/cache.stage2/cookbooks/docker/libraries/provider_docker_service_execute.rb:1:in `<top (required)>'
 
 
Relevant File Content:
----------------------
/var/lib/aws/opsworks/cache.stage2/cookbooks/docker/libraries/provider_docker_service_execute.rb:
 
1:  class Chef
2:    class Provider
3:      class DockerService
4:        class Execute < Chef::Provider::DockerService
5>>         provides :docker_service, os: ' linux'
6:  
7:          # Start the service
8:          action :start do
9:            # Go doesn't support detaching processes natively, so we have
10:            # to manually fork it from the shell with &
11:            # https://github.com/docker/docker/issues/2758
12:            bash 'start docker' do
13:              code "#{docker_daemon_cmd} &>> #{docker_log} &"
14:              environment 'HTTP_PROXY' => new_resource.http_proxy,

**Comment by seanadkinson** [comment]
Getting the same error all of a sudden.

**Comment by archen** [comment]
also receiving this error.

**Comment by scalp42** [comment]
@seanadkinson @archen @josephsiefers which version of Chef are you guys running ?

**Comment by seanadkinson** [comment]
@scalp42 I'm running on AWS OpsWorks, Chef version 11.10.

**Comment by josephsiefers** [comment]
@scalp42, same here, AWS OpsWorks, Chef version 11.10

**Comment by archen** [comment]
@scalp42 running from OS X 10.10 to provision an Ubuntu node.

Chef Development Kit Version: 0.6.2
chef-client version: 12.3.0
berks version: 3.2.4
kitchen version: 1.4.0

**Comment by ajohnstone** [comment]
As above...
```

================================================================================
Recipe Compile Error in /var/lib/aws/opsworks/cache.stage2/cookbooks/docker/libraries/provider_docker_service_execute.rb
================================================================================
 
 
NoMethodError
-------------
undefined method `provides' for Chef::Provider::DockerService::Execute:Class
 
 
Cookbook Trace:
---------------
/var/lib/aws/opsworks/cache.stage2/cookbooks/docker/libraries/provider_docker_service_execute.rb:5:in `<class:Execute>'
/var/lib/aws/opsworks/cache.stage2/cookbooks/docker/libraries/provider_docker_service_execute.rb:4:in `<class:DockerService>'
/var/lib/aws/opsworks/cache.stage2/cookbooks/docker/libraries/provider_docker_service_execute.rb:3:in `<class:Provider>'
/var/lib/aws/opsworks/cache.stage2/cookbooks/docker/libraries/provider_docker_service_execute.rb:2:in `<class:Chef>'
/var/lib/aws/opsworks/cache.stage2/cookbooks/docker/libraries/provider_docker_service_execute.rb:1:in `<top (required)>'
 
 
Relevant File Content:
----------------------
/var/lib/aws/opsworks/cache.stage2/cookbooks/docker/libraries/provider_docker_service_execute.rb:
 
1:  class Chef
2:    class Provider
3:      class DockerService
4:        class Execute < Chef::Provider::DockerService
5>>         provides :docker_service, os: ' linux'
6:  
7:          # Start the service
8:          action :start do
9:            # Go doesn't support detaching processes natively, so we have
10:            # to manually fork it from the shell with &
11:            # https://github.com/docker/docker/issues/2758
12:            bash 'start docker' do
13:              code "#{docker_daemon_cmd} &>> #{docker_log} &"
14:              environment 'HTTP_PROXY' => new_resource.http_proxy,
 
```

**Comment by jmasson-coveo** [comment]
Same thing for me!

Does anyone have a solution? 

================================================================================
Recipe Compile Error in /var/lib/aws/opsworks/cache.stage2/cookbooks/docker/libraries/provider_docker_service_execute.rb
================================================================================
 
 
NoMethodError
-------------
undefined method `provides' for Chef::Provider::DockerService::Execute:Class
 
 
Cookbook Trace:
---------------
/var/lib/aws/opsworks/cache.stage2/cookbooks/docker/libraries/provider_docker_service_execute.rb:5:in `<class:Execute>'
/var/lib/aws/opsworks/cache.stage2/cookbooks/docker/libraries/provider_docker_service_execute.rb:4:in `<class:DockerService>'
/var/lib/aws/opsworks/cache.stage2/cookbooks/docker/libraries/provider_docker_service_execute.rb:3:in `<class:Provider>'
/var/lib/aws/opsworks/cache.stage2/cookbooks/docker/libraries/provider_docker_service_execute.rb:2:in `<class:Chef>'
/var/lib/aws/opsworks/cache.stage2/cookbooks/docker/libraries/provider_docker_service_execute.rb:1:in `<top (required)>'
 
 
Relevant File Content:
----------------------
/var/lib/aws/opsworks/cache.stage2/cookbooks/docker/libraries/provider_docker_service_execute.rb:
 
1:  class Chef
2:    class Provider
3:      class DockerService
4:        class Execute < Chef::Provider::DockerService
5>>         provides :docker_service, os: ' linux'
6:  
7:          # Start the service
8:          action :start do
9:            # Go doesn't support detaching processes natively, so we have
10:            # to manually fork it from the shell with &
11:            # https://github.com/docker/docker/issues/2758
12:            bash 'start docker' do
13:              code "#{docker_daemon_cmd} &>> #{docker_log} &"
14:              environment 'HTTP_PROXY' => new_resource.http_proxy,

**Comment by josephsiefers** [comment]
0.37.0 works for me for now.

**Comment by rtrauntvein** [comment]
Seems like 0.40 introduced a lot of breaking changes.  #338

**Comment by okram999** [comment]
0.40.0 broke the chef run for me as well... This is complaining of the syntax error. Is this because i am on chef-client 11.16.4?


----------------------------



Starting Chef Client, version 11.16.4
Compiling Cookbooks...

================================================================================
Recipe Compile Error in /home/centos/chef-solo/cookbooks-2/docker/libraries/provider_docker_service_execute.rb
================================================================================

NoMethodError
-------------
undefined method `provides' for Chef::Provider::DockerService::Execute:Class

Cookbook Trace:
---------------
  /home/centos/chef-solo/cookbooks-2/docker/libraries/provider_docker_service_execute.rb:5:in `<class:Execute>'
  /home/centos/chef-solo/cookbooks-2/docker/libraries/provider_docker_service_execute.rb:4:in `<class:DockerService>'
  /home/centos/chef-solo/cookbooks-2/docker/libraries/provider_docker_service_execute.rb:3:in `<class:Provider>'
  /home/centos/chef-solo/cookbooks-2/docker/libraries/provider_docker_service_execute.rb:2:in `<class:Chef>'
  /home/centos/chef-solo/cookbooks-2/docker/libraries/provider_docker_service_execute.rb:1:in `<top (required)>'

Relevant File Content:
----------------------
/home/centos/chef-solo/cookbooks-2/docker/libraries/provider_docker_service_execute.rb:

  1:  class Chef
  2:    class Provider
  3:      class DockerService
  4:        class Execute < Chef::Provider::DockerService
  5>>         provides :docker_service, os: ' linux'
  6:  
  7:          # Start the service
  8:          action :start do
  9:            # Go doesn't support detaching processes natively, so we have
 10:            # to manually fork it from the shell with &
 11:            # https://github.com/docker/docker/issues/2758
 12:            bash 'start docker' do
 13:              code "#{docker_daemon_cmd} &>> #{docker_log} &"
 14:              environment 'HTTP_PROXY' => new_resource.http_proxy,


Running handlers:
[2015-07-13T17:29:46+00:00] ERROR: Running exception handlers
Running handlers complete
[2015-07-13T17:29:46+00:00] ERROR: Exception handlers complete
[2015-07-13T17:29:46+00:00] FATAL: Stacktrace dumped to /var/chef/cache/chef-stacktrace.out
Chef Client failed. 0 resources updated in 0.667109141 seconds
[2015-07-13T17:29:46+00:00] ERROR: undefined method `provides' for Chef::Provider::DockerService::Execute:Class
[2015-07-13T17:29:47+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
ERROR: RuntimeError: chef-solo failed. See output above.

**Comment by someara** [comment]
This is fixed up as of v0.40.3
